src/riak_kv_bucket.erl

```diff
@@ -540,16 +564,95 @@ validate_last_write_wins_implies_not_dvv_enabled({Props, Errors}) ->
             {Props, Errors}
     end.
 
-%% @private just grab the allow_mult value if it exists
--spec allow_mult(props()) -> boolean() | 'undefined' | 'error'.
-allow_mult(Props) ->
-    case proplists:get_value(allow_mult, Props) of
+%% @doc See if datatype is valid, if so return the datatype, otherwise
+%%      false for handling.
+-spec datatype(props()|datatype_names(), props()|datatype_names()) ->
+                      datatype_names() | false.
+datatype(PropsNew, PropsOld) when is_list(PropsOld), is_list(PropsNew) ->
+    datatype(proplists:get_value(datatype, PropsNew),
+             proplists:get_value(datatype, PropsOld));
+datatype(DataTypeNew, undefined) ->
+    case lists:member(riak_kv_crdt:to_mod(DataTypeNew), ?V2_TOP_LEVEL_TYPES) of
```

Comment:
I know this is the same as released, but I wonder if it is right. Should we be using the capability negotiated list, for example? Also, see comment on `riak_ky_types.hrl` diff: I think this new CRDT (HLL) changes our version/capabilities list, etc.